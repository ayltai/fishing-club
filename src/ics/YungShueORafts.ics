BEGIN:VCALENDAR
VERSION:1.0
BEGIN:VEVENT
DTSTART:20170709T113000
DTEND:20170709T183000
LOCATION:Yung Shue O Fish Cultivation Rafts, Sai Kung
DESCRIPTION:
SUMMARY:Yung Shue O Rafts
PRIORITY:3
END:VEVENT
END:VCALENDAR
