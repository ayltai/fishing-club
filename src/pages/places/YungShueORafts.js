// @flow
'use strict';

import React from 'react';
import muiThemeable from 'material-ui/styles/muiThemeable';
import { Card, CardMedia, CardTitle, CardText } from 'material-ui/Card';
import LazyLoad from 'react-lazy-load';
import './YungShueORafts.css';
import OurRaft1Image from '../../images/our-raft-1.jpg';
import OurRaft2Image from '../../images/our-raft-2.jpg';
import OurRaft3Image from '../../images/our-raft-3.jpg';
import RaftFloatingImage from '../../images/raft-floating.jpg';

class YungShueORafts extends React.Component {
    componentDidMount() : void {
        window.scrollTo(0, 0);
    }

    render() : any {
        return (
            <div>
                <Card className="yungShueORaftItem">
                    <CardMedia>
                        <img
                            src={RaftFloatingImage}
                            style={{
                                maxWidth : '100%'
                            }} />
                    </CardMedia>
                    <CardTitle title="Yung Shue O Rafts (榕樹澳漁排)" />
                    <CardText>
                        <p>Yung Shue O fish cultivation facilities have become sites for recreational activities since 15 years ago (2002) as one of the steps the Government planned to promote local economic growth. There are over a hundred fish cultivation rafts. Some of them feature water sports, some feature fishing in closed areas where the fish is trapped. The one we chosed is 魚樂會, where we are going to fish in the middle of the ocean!</p>
                        <p>A fish cultivation raft is a floating structure, built using woods with large empty plastic cylinder containers underneath. The structure is so large that you can hardly feel you are floating on water.</p>
                    </CardText>
                </Card>
                <Card className="yungShueORaftItem">
                    <CardTitle title="Our raft" />
                    <CardText>
                        <LazyLoad
                            width={768}
                            height={640}>
                            <div
                                style={{
                                    textAlign : 'center'
                                }}>
                                <img
                                    src={OurRaft1Image}
                                    style={{
                                        maxWidth : '100%'
                                    }} />
                            </div>
                        </LazyLoad>
                        <br />
                        <LazyLoad
                            width={768}
                            height={400}>
                            <div
                                style={{
                                    textAlign : 'center'
                                }}>
                                <img
                                    src={OurRaft2Image}
                                    style={{
                                        maxWidth : '100%'
                                    }} />
                            </div>
                        </LazyLoad>
                        <br />
                        <LazyLoad
                            width={768}
                            height={400}>
                            <div
                                style={{
                                    textAlign : 'center'
                                }}>
                                <img
                                    src={OurRaft3Image}
                                    style={{
                                        maxWidth : '100%'
                                    }} />
                            </div>
                        </LazyLoad>
                    </CardText>
                </Card>
            </div>
        );
    }
}

export default muiThemeable()(YungShueORafts);
