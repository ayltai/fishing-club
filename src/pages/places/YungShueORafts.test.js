'use strict';

import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import YungShueORafts from './YungShueORafts';

it('renders correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <BrowserRouter>
            <MuiThemeProvider>
                <YungShueORafts />
            </MuiThemeProvider>
        </BrowserRouter>,
        div);

    expect(div.innerHTML).toMatchSnapshot();
});
