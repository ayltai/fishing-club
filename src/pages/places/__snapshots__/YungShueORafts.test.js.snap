// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `

<div data-reactroot>
  <div class="yungShueORaftItem"
       style="color: rgba(0, 0, 0, 0.87); background-color: rgb(255, 255, 255); box-sizing: border-box; font-family: Roboto, sans-serif; box-shadow: 0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12); border-radius: 2px; z-index: 1;"
  >
    <div style="padding-bottom: 8px;">
      <div style="position: relative;">
        <div>
          <img src="raft-floating.jpg"
               style="vertical-align: top; max-width: 100%; min-width: 100%; width: 100%;"
          >
        </div>
        <!-- react-text: 7 -->
        <!-- /react-text -->
      </div>
      <div style="padding: 16px; position: relative;">
        <span style="font-size: 24px; color: rgba(0, 0, 0, 0.87); display: block; line-height: 36px;">
          Yung Shue O Rafts (榕樹澳漁排)
        </span>
        <span style="font-size: 14px; color: rgba(0, 0, 0, 0.54); display: block;">
        </span>
      </div>
      <div style="padding: 16px; font-size: 14px; color: rgba(0, 0, 0, 0.87);">
        <p>
          Yung Shue O fish cultivation facilities have become sites for recreational activities since 15 years ago (2002) as one of the steps the Government planned to promote local economic growth. There are over a hundred fish cultivation rafts. Some of them feature water sports, some feature fishing in closed areas where the fish is trapped. The one we chose is 魚樂會, where we are going to fish in the middle of the ocean!
        </p>
        <p>
          A fish cultivation raft is a floating structure, built using woods with large empty plastic cylinder containers underneath. The structure is so large that you can hardly feel you are floating on water.
        </p>
      </div>
    </div>
  </div>
  <div class="yungShueORaftItem"
       style="color: rgba(0, 0, 0, 0.87); background-color: rgb(255, 255, 255); box-sizing: border-box; font-family: Roboto, sans-serif; box-shadow: 0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12); border-radius: 2px; z-index: 1;"
  >
    <div style="padding-bottom: 8px;">
      <div style="padding: 16px; position: relative;">
        <span style="font-size: 24px; color: rgba(0, 0, 0, 0.87); display: block; line-height: 36px;">
          The event
        </span>
        <span style="font-size: 14px; color: rgba(0, 0, 0, 0.54); display: block;">
        </span>
      </div>
      <div style="padding: 16px; font-size: 14px; color: rgba(0, 0, 0, 0.87);">
        <p>
          <!-- react-text: 21 -->
          The event will be held on
          <!-- /react-text -->
          <strong>
            July 9, 2017 (Sunday)
          </strong>
          <!-- react-text: 23 -->
          , from noon till evening.
          <!-- /react-text -->
        </p>
        <p>
          <!-- react-text: 25 -->
          Reserve your day by importing this iCalendar appointment file to your device:
          <!-- /react-text -->
          <br>
          <span style="font-size: xx-large;">
            <a href="YungShueORafts.ics"
               title="iCalendar appointment"
            >
              <i class="icofont icofont-ui-calendar">
              </i>
            </a>
          </span>
        </p>
        <div class="LazyLoad"
             style="height: 640px; width: 768px;"
        >
        </div>
        <br>
        <div class="LazyLoad"
             style="height: 400px; width: 768px;"
        >
        </div>
        <br>
        <div class="LazyLoad"
             style="height: 400px; width: 768px;"
        >
        </div>
      </div>
    </div>
  </div>
</div>

`;
